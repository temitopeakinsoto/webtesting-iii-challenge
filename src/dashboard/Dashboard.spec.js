// Test away
import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Dashboard from './Dashboard';

afterEach(rtl.cleanup);

let wrapper

beforeEach(() => {
  wrapper = rtl.render(<Dashboard />);
});

describe('Dashboard component', () => {
    it('dashboard component matches the snapshot', () => {
        expect(wrapper.container).toMatchSnapshot();
    });
    it('dashboard component to have a lock gate button', () => {
        expect(wrapper.queryByText(/Lock Gate/i)).toBeInTheDocument();
    });
})