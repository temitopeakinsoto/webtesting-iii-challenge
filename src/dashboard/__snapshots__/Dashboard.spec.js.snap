// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component dashboard component matches the snapshot 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</div>
`;

exports[`Dashboard component display and controls dashboard component matches the snapshot 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</div>
`;

exports[`Dashboard component displays the dashboard component 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</div>
`;
